internal/addon/authentication/certificates.go

```diff
@@ -24,24 +24,34 @@ func CreateOrUpdateRootCertificate(k8s client.Client) error {
 
 	objects := manifests.BuildAllRootCertificate()
 
+	var errCount int32
 	for _, obj := range objects {
+		l := log.WithValues(
+			"object_name", obj.GetName(),
+			"object_kind", obj.GetObjectKind().GroupVersionKind().Kind,
+		)
+
 		desired := obj.DeepCopyObject().(client.Object)
 		mutateFn := manifests.MutateFuncFor(obj, desired, nil)
 
 		op, err := ctrl.CreateOrUpdate(ctx, k8s, obj, mutateFn)
 		if err != nil {
-			klog.Error(err, "failed to configure resource")
+			l.Error(err, "failed to configure resource")
+			errCount++
 			continue
 		}
 
-		msg := fmt.Sprintf("Resource has been %s", op)
+		msg := fmt.Sprintf("resource has been %s", op)
 		switch op {
 		case ctrlutil.OperationResultNone:
-			klog.Info(msg)
+			l.Info(msg)
 		default:
-			klog.Info(msg)
+			l.Info(msg)
 		}
 	}
+	if errCount > 0 {
```

Comment:
You can just enable a flag or initialize an `err` variable, right?

Reply:
I've added the errorCount to the message